.github/workflows/on-demand.yml

```diff
@@ -0,0 +1,33 @@
+name: OnDemand Lunatic CI (publish every branch)
+on:
+  workflow_dispatch:
+jobs:
+  publish_on_npm:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

Comment:
Maybe

```
      - name: Checkout code
        uses: actions/checkout@v3
        with:
          ref: ${{ github.ref }}
```

if you want to checkout the triggering branch (I may be wrong but seems to me that checkout will have main/master as default)